### ISSUE TITLE
Log rotate sidecar should run as ceph user 

### ISSUE THREAD

**Issue by travisn** [rook/rook#11235]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The log rotate container needs to create the security context for the ceph user to avoid running as root.

**Expected behavior:**
All ceph daemon containers are expected to run as the ceph user id 167.

**Comment by travisn** [comment]
Completed with #11219

**Comment by travisn** [comment]
Actually, #11219 just addressed the crash collector

**Comment by travisn** [comment]
After working through this, running the logrotate container as the ceph user is complicated with the following:
- The `/etc/logrotate.d/ceph` config file is from the base ceph container. The log rotate container modifies the config before running `logrotate`. If we need to modify this file, an init container needs to chown that file, copy it to a temporary directory in an empty directory volume, then re-mount the empty directory volume in the main container.
- `logrotate` requires access to some directories such as `/etc/logrotate.d` and `/var/lib/logrotate/`, which would need to be chown'd

Even after prototyping the above changes to run as the ceph user `167`, `logrotate` gives an error:
```
rotating pattern: /var/log/ceph/ceph-mon.a.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
error: error switching euid to 0 and egid to 167: Operation not permitted
```
Given the complexity of setting this up, we need to ask the question about what is the security risk of continuing to run logrotate as `root` inside the container. The similar risk for the crash collector addressed by #11219 was that a user with access to the host could get the crash collector to upload modified files. In the case of logrotate, all we are doing is rotating the log files. If the user has access to the host, they aren't going to accomplish anything except delete or modify the log files, which are never uploaded anywhere. 

If no security risk is identified, I recommend we keep the logrotate container implementation simple and not change this behavior.
